sycl/source/detail/pi.cpp

```diff
@@ -131,6 +150,55 @@ vector_class<plugin> initialize() {
   // TODO: Correct the logic to store the appropriate plugin into GlobalPlugin
   // variable. Currently it saves the last plugin found.
   GlobalPlugin = std::make_shared<plugin>(PluginInformation);
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Not sure this is the best place to initialize
+  // the framework; SYCL runtime team needs to
+  // advise on the right place, until then we
+  // piggy-back on the initialization of the PI layer.
+  //
+  if (xptiTraceEnabled()) {
+    // Registers a new stream for 'sycl' and any
+    // plugin that wants to listen to this stream
+    // will register itself using this string or
+    // stream ID for this string.
+    //
+    auto StreamID = xptiRegisterStream(SYCL_STREAM_NAME);
+    //  Let all tool plugins know that a stream
+    //  by the name of 'sycl' has been initialized
+    //  and will be generating the trace stream.
+    //
+    //                                  +--- Minor version #
+    //            Major version # ---+  |   Version string
+    //                               |  |       |
+    //                               v  v       v
+    xptiInitialize(SYCL_STREAM_NAME, 1, 0, "sycl 1.0");
+    // Create a tracepoint to indicate the graph
+    // creation
+    //
+    xpti::payload_t p("application_graph");
+    auto Event = xptiMakeEvent("application_graph", &p,
+                               (uint16_t)xpti_te::graph, xpti_at::active);
+    if (Event) {
```

Comment:
I guess Event has a pointer type, so what if PIEvent is nullptr here?

Reply:
All future notifications that require the graph event parent will return immediately - no graph information will be generated. Only the trace information will be generated.